driftbase/models/db.py

```diff
@@ -143,11 +169,18 @@ def __marshallable__(self):
 
     @hybrid_property
     def is_online(self):
+        """
+        Deprecated. Use rich_presence.is_online instead.
+        """
         if self.user and self.user.client:
             return self.user.client.is_online
         return False
 
+    @hybrid_property
```

Comment:
I'm not convinced that putting arbitrary queries for related data into the core player is the way to go, especially since it uses `g.db` directly, which makes it harder to mock and test this logic. Admittedly the layering in drift-base is very poor, so this may not be a real-world issue until we rewrite the whole thing, but I still feel that it would make more sense to query for this explicitly where it is needed.

If there was a relationship between core players and match players perhaps, but I don't think such a relationship should be on the model, since they are unrelated concepts, and candidates for being refactored into separate applications.

Reply:
Thanks for the feedback! 

Just to clarify my reasoning: In Perseus we access `is_online` field by querying for friends, and then filtering only for the 'is_online' column. We poll these values every few seconds. This existing setup is what inspired me to stick rich_presence into the player, so we can keep the logic in Perseus largely the same. 

**Proposed alternative:** I can introduce a completely separate flask endpoint called something like `get_rich_presence`, and return the `g.db` calculated data directly. This also allows me to toss some boilerplate related to Marshmallow (at least I think so). Thoughts?

Reply:
Implemented something along the lines of what I described. If you have a different design idea, just let me know :)

Currently, what's missing from my implementation is an easy way to grab the status for a lot of related players. We used to get this for "free" from the players endpoint, since we could filter players by friend status, and then filter also the db rows returned by key. 

Now that I moved rich presence into it's own endpoint, we lack this flexibility. I can add something like `richpresence.list` (see `players.list`), but it would require me to re-implement the filtering options defined there. Not too bad to do, but I wanted to hear your opinions on the direction first.

Reply:
You could have root resources /[rich_]presence[?player_ids=[...]] and /[rich_]presence/player-id since this could be thought of as its own resource. 

It does seem odd to have the client ask for this every few seconds though. The pattern ought to be:

1. Ask for the current state of all relevant players
2. Listen for updates on the messages endpoint

I think the notifications are probably not being sent to all relevant players now, and that's what's preventing this from being as nice as it could be, so looking into how that can be improved, if it's possible or won't work for all cases, would be a good idea.

But yes, having a way to query for more than one player at a time is definitely worth it compared to bombarding the backend with multiple requests. In fact, Unreal is already complaining that we're sending too many requests in parallel

Reply:
Richpresence has been migrated into it's own flask extension. 

Additionally, rich presence is no longer *polled* at all, but rather kept up to date in redis by listening to various events (e.g., player join match, player leave match). 

It's now possible to query information (via flask endpoint), or listen to the message queue 'richpresence' in order to get updates without polling.